iothub/device/src/MethodRequest.cs

```diff
@@ -28,18 +54,37 @@ public MethodRequest(string name, byte[] data, TimeSpan? responseTimeout, TimeSp
             ConnectionTimeout = connectionTimeout;
         }
 
+        /// <summary>
+        /// The method name.
+        /// </summary>
         public string Name { get; private set; }
 
+        /// <summary>
+        /// The method data.
+        /// </summary>
         [System.Diagnostics.CodeAnalysis.SuppressMessage(
             "Performance",
             "CA1819:Properties should not return arrays",
             Justification = "Cannot change property types on public classes.")]
         public byte[] Data { get; private set; }
 
+        /// <summary>
+        /// The method response timeout value. This value is the amount of time that IoT Hub service must await for
+        /// completion of a direct method execution on a device. The minimum and maximum values are 5 and 300 seconds.
+        /// </summary>
```

Comment:
it is worth specifying here that these values for `MethodRequest` are relevant only when invoking methods from one edge module to another. For device client operations the user will receive the `MethodRequest` as a part of the direct method callback invoked, so they wouldn't really be setting these timeout values.